File: server/src/main/kotlin/com/coooolfan/xiaomialbumsyncer/xiaomicloud/CloudSpaceInfo.kt
package com.coooolfan.xiaomialbumsyncer.xiaomicloud

/**
 * 云端空间使用详情项
 */
data class SpaceUsageItem(
    val size: Long,     // 使用空间（字节）
    val text: String    // 显示文本
)

/**
 * 云端空间使用情况信息
 */
data class CloudSpaceInfo(
    val totalQuota: Long,     // 总空间（字节）
    val used: Long, // 已用空间（字节）
    val galleryUsed: Long, // 相册使用空间（字节）
    val usagePercent: Int, // 使用百分比
    val usedDetail: Map<String, SpaceUsageItem> // 详细使用信息
)

File: web/src/__generated/model/static/CloudSpaceInfo.ts
import type {SpaceUsageItem} from './';

/**
 * 云端空间使用情况信息
 */
export interface CloudSpaceInfo {
    readonly totalQuota: number;
    readonly used: number;
    readonly galleryUsed: number;
    readonly usagePercent: number;
    readonly usedDetail: {readonly [key:string]: SpaceUsageItem};
}
